### ISSUE TITLE
Backup of all settings

### ISSUE THREAD

**Issue by anpic** [emanuele-f/PCAPdroid#379]
I will also share my practice.
Since 31 sdk version Google has killed another very useful thing: local backup and restore settings.
And your target sdk version is 33.
Accordingly for this users have to rebuild your beautiful app. And it's not nice to do that :)
Maybe you'll also think about adding full importing and exporting settings through intents. Including geolocation databases.

**Comment by emanuele-f** [comment]
Possibly via https://developer.android.com/reference/android/app/backup/SharedPreferencesBackupHelper

**Comment by anpic** [comment]
> Possibly via https://developer.android.com/reference/android/app/backup/SharedPreferencesBackupHelper

That would be great :)
But I would like not only the settings but also files as geodb.
But at least to start with something ;)

**Comment by emanuele-f** [comment]
>But I would like not only the shared preferences but also files such as geodb

Geoip can be easily downloaded and is MB in size, only prefs and similar stuff will be backed up

**Comment by anpic** [comment]
> Geoip can be easily downloaded and is MB in size

How can I automatically add and update this database without manual actions? ;)

**Comment by anpic** [comment]
I think you're making the same mistake as Marcel ;)
You're targeting dumb users :)
Dumb users don't need firewalls. Or are firewalls sufficient for them such as https://github.com/StarGW-net/karma-firewall
And the users of your beautiful firewall need intents to automate actions ;)

**Comment by emanuele-f** [comment]
@anpic please realize you are flooding me with very specific questions, which have little impact on the overall usability and value of the app. Kindly ask you to focus on the important stuff, otherwise I won't be able to read everything and concentrate on the development. If you really want to help me, pick an open issue, implement it and provide a pull request :)

**Comment by emanuele-f** [comment]
> Then you can placidly choose priorities, but without closing all the issues in a row

You have opened them in a row and so far you have not accepted a "no" as a reply, so closing them is a necessity for me to keep the issues list under a sane limit that I can manage.
An issue cannot be the first thing that comes to your mind (that can be a discussion on telegram/new Discussion). To bring value, it should provide as much info as possible (even a preliminary design), explaining the use case, which should cover a real necessity for most users. 
Even in this case, I will close it if I think it does not really meet the project goals.
Having a limited time to dedicate to this project, I have to filter out the noise and focus on what's more important for the project.

Ps. Merry Christmas!

**Comment by anpic** [comment]
> To bring value, it should provide as much info as possible (even a preliminary design), explaining the use case, which should cover a real necessity for most users

But you're not even trying to understand what I'm writing anymore, you're not trying to discuss it, you're trying to brush it off as quickly as possible.
So it makes sense to provide everything right away if I see that you are determined to "freeze" the project? ;)
I'm not a programmer but I understand a lot about firewalls and android. I've probably tried and tested all kinds of firewalls for android.

> An issue cannot be the first thing that comes to your mind

I understand more about firewalls than your entire community. Consider it bragging. 
I'm ready to discuss everything in a substantive and reasoned manner. It's here where can split the issues by topic. And not in messengers where crowds of stupid users sit, often not understanding even simple things. And where all the discussions are mixed into a single mess.
But I see that you don't have time to discuss and develop the project.

**Comment by emanuele-f** [comment]
>But you're not even trying to understand what I'm writing anymore, you're not trying to discuss it, you're trying to brush it off as quickly as possible.

Actually I'm trying and I've kept open the issues which are relevant for the project. I prefer short replies and getting to the point, to save my and other people time. Unfortunately we are time bound as human beings, so we should make the best use of our time

>So it makes sense to provide everything right away if I see that you are determined to "freeze" the project? ;)
>I understand more about firewalls than your entire community.

These sentences alone explain the whole problem in your attitude, which is why you are not bringing much value to this/other projects. You think you are the only one who knows how the things work and that your opinion is superior to anyone else, even the developers of a project. You are not, and the only way to realize this it to build your product from the ground up and try to bring it to the masses. Only in this way you can grasp what is the effort required to build even a simple feature like this backup one, and hopefully realize we only know a very small amount of how things in world work

**Comment by anpic** [comment]
> Actually I'm trying and I've kept open the issues which are relevant for the project

Or are you kidding yourself? ;)
Look at the number of your commits in the past years and this year ;)
I suggest you just call a spade a spade. The absence of time by the absence of time. But you try to create psychological comfort for yourself and quickly close everything down. Although in reality this is the freezing of the project.
And in this way you are rejecting possible competent people from the project.
You never know what will happen in the future to me, to you, but if our technical discussions remain, they can help others who suddenly want to continue your great project. The AFWall project understands this perfectly well, and doesn't try to close issues just like that.

**Comment by anpic** [comment]
> These sentences alone explain the whole problem in your attitude, which is why you are not bringing much value to this/other projects

If I'm not a programmer, how do you suggest I create everything from the ground up?
But I'm someone who understands exactly what a firewall is, and what functions it can have so that it meets world standards.
It's funny that the guys from Rethink reopen my issues, although I specifically close them. And they even transfer it from your project.
But they have finances, desire, time and goals that may not be as good as they seem :(

**Comment by anpic** [comment]
Anyway, I repeat once again, you are a really cool developer. Not as great as Marcel yet, but cool :)
And you've created a great project.
But don't fool yourself and others. Let the project develop as it is, don't close issues without careful study and understanding, just to create your own psychological comfort.
And maybe someday, when you have time for a project again, you will understand how important it is not to make the project hostage to your psychological comfort ;)

**Comment by anpic** [comment]
> we only know a very small amount of how things in world work

That's why you need to learn to listen to others, and not close their mouths ;)
But this is your project, your brainchild, it's up to you to decide how he lives on.
Happy holidays :)

**Comment by emanuele-f** [comment]
>In AFWall project understands this perfectly well, and doesn't try to close issues just like that.

271 open issues, many of them without even a reply... Is this a sign of good project health 🤔 ?

>And maybe someday, when you have time for a project again, you will understand how important it is not to make the project hostage to your psychological comfort ;)

Again, it's not you who has the knowledge of how things work, of when a developer is willing to develop or not. Talking like that is just a sign of arrogance, and this no way can help projects grow. So please stop talking to satisfy your own ego and draining developers energies and time with that. You now know how you can help if you really want to

**Comment by anpic** [comment]
> Is this a sign of good project health 🤔 ?

Are you sure that everything is all right in your kingdom? And even more so, how will it be next year? ;)

> So please stop talking to satisfy your own ego

So far, I'm opening issues with suggestions for development, understanding exactly what it's and why, and you're closing them without even figuring it out. While others are taking mine away and reopening it. That's the reality.
There is no ego here. I would help to help turn the project into the best of all time.
But you don't have time for him. And instead of admitting it, you're already trying to pull my ego, which isn't there, too :)

**Comment by emanuele-f** [comment]
>While others are taking mine away and reopening it.

No, this trick won't work here. Seems like @ignoramous fell into your trap, where you referenced my name and project to justify your wrong assumptions and bad behavior. I'm don't want to spend 1 minute more with you. Your intention is clearly not to grow the projects but your own ego

**Comment by anpic** [comment]
> Your intention is clearly not to grow the projects but your own ego

It's your full right. To think so or pretend to think so.
I hope that the project will develop after all, and not stand still.
And I hope to meet here in a year with a lot of new things implemented ;)
And I will continue to amuse my ego alone :)
Have a nice holidays and good luck in life!

**Comment by ignoramous** [comment]
> No, this trick won't work here. Seems like @ignoramous fell into your trap

Hi Emanuele: I left you an apology on telegram but unsure you saw it before it was deleted.

What I pointed out was that if *anyone* thinks a userspace app (*root* or otherwise) can block ALL traffic on Android, then they're ignorant about Android's internals, that is, not familiar with how Android works. I'm not sure why you'd take this statement as an insult (from a conversation where I didn't tag you at all). Anyways, it is never my intention to downplay any project or developer (especially, something as impressive as pcapdroid). And I apologise unconditionally if that or any other comment came off as an insult.

**Comment by anpic** [comment]
You're on time :)
But instead of all these empty words, it's better to join forces and finally create a powerful and functional firewall.
Sometimes it feels like I got into kindergarten, and it's not the users and successes in creating collective security that are important to yours, but your own sense of psychological comfort. Marcel burned out like that. I hope Emanuele turns out to be stronger.

> What I pointed

You also said a few other things: that Emanuele doesn't respect android internals. Hinting at how traffic is handled inside PCAPdroid's VpnService. But inside your Rethink and Firestack there is still a modification of connections, albeit much deeper. So why these discussions?
If you all really want to achieve the goals you claim, join forces and finally create a LittleSnitch-class product. 
And I won't have to offer my stupid ideas. And I will finally be able to amuse my ego in silence and oblivion :))

**Comment by emanuele-f** [comment]
>Hi Emanuele: I left you an apology on telegram but unsure you saw it before it was deleted.

Hello, that was deleted to avoid bringing problems to the PCAPdroid group. I planned to contact you later, I've just sent you a message to get in touch

>What I pointed out was that if anyone thinks a userspace app (root or otherwise) can block ALL traffic on Android, then they're ignorant about Android's internals, that is, not familiar with how Android works

You said I was ignorant (you wrote my name) of how Android works, but I did not participate in that conversation at all, that was the problem. However I notice that @anpic has questionable ways to support his opinions, which involves taking in other projects/developers, so I understand the situation and this topic is closed for me, no problem

>from a conversation where I didn't tag you at all

This was directed to @anpic

>If you all really want to achieve the goals you claim, join forces and finally create together a LittleSnitch-class product.

This is not something that can happen from day to day, it requires sharing objectives, visions and agreement. Also sometimes it's still good to provide people some alternatives

Anyway, we are again using a github task to discuss unrelated things, I suggest again to use other channels like telegram or the Github discussions

**Comment by anpic** [comment]
> However I notice that @anpic has questionable ways to support his opinions

Is it a questionable way to involve top-level professionals like you to discuss relevant issues?
Are donations and in-app purchases still not questionable ways for us to support your developments? ;)
Guys, in my opinion, you stayed too long in your kingdoms, and stopped noticing smart users, betting only on stupid ones.
Of course, it's up to yours who I am.
But I just tried to remind you that dumb users don't need firewalls and smart users need smart firewalls.
I've tried a lot of firewalls in recent years, and specifically for android probably everything that exists at all.
And I thought why not give the community at least a little bit of my experience and knowledge. And I chose, in my opinion, some of the best and at least somehow developing ones.
But it was in the project that I consider the most cool and promising that my opinion turned out to be unnecessary. Because Emanuele seems to think that he knows everything, and he only needs other developers to help him, and the user experience is needed only from stupid users.
Perhaps this is true, and I myself believe that Marcel and Emanuele are the best android network stack specialists in the world.
But why surround yourself with just dumb users? Who aren't able to say a word of arguments in favor of other points of view on various technical issues.
Yes, it is convenient with such people, they will take everything for granted, they will not say a word for a reasoned answer.
And with me, yes, it will not be easy, but every suggestion I make will be supported by weighty arguments.
If the reason is only psychological comfort, then fine, I will no longer disturb and create "unnecessary" issues.
But my desired goal is to see the creation of a full-fledged android firewall in this life :)

> Also sometimes it's still good to provide people some alternatives

Yes, especially when in one of them the developers do not know that they themselves understand what they are doing, in the other there are no useful things from the first.
And in Android is only one VpnService there :)

> or the Github discussions

What is the point of them without the participation of the main heroes: you as developers? ;)
You yourself consider them places where you can not only not answer but not even look

**Comment by emanuele-f** [comment]
>Because Emanuele seems to think that he knows everything

No, I don't think so. But a project must have a direction, which involves making choices

>But why surround yourself with just dumb users

If the target was the dumb user, than features like the TLS decryption would not be there. However, to provide a nice user experience, it's better to think like a dumb user than a tech savvy.

>What is the point of them without the participation of the main heroes: you as developers? ;)
>You yourself consider them places where you can not only not answer but not even look

I never said I would not participate. It's important to keep things tidy, and an issue does not seem a good place to discuss this. E.g. I've just open https://github.com/emanuele-f/PCAPdroid/discussions/391 When everything is clear, a well defined, specific issue can be open for implementation. So please let's move these kind of discussions there, I'm getting tired of hiding comments
